### PR TITLE
Added a new error state message the status 404

### PR DESCRIPTION
**Pull request by Fewwy**
1)Created a new component NoRecsState similar to the MessageState
2)Added a step to the Cluster.js to check if the error is true and have a code is equal to 404 to render new component NoRecsState
3)Created new error messages and descriptions and added them to the Messages.js

image

### CODE REVIEW

File: src/Components/MessageState/NoRecsState.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  icon,
+  iconClass,
+  iconStyle,
+  text,
+  title,
+  variant,
+  link,
+  linktext,
+}) => (
+  <EmptyState className={className} variant={variant}>
```
Comment (gkarat) [inline src/Components/MessageState/NoRecsState.js:25]: Any reason to keep it separately? Looks like can be added to https://github.com/RedHatInsights/ocp-advisor-frontend/blob/master/src/Components/MessageState/EmptyStates.js

File: src/Components/Cluster/Cluster.js [hunk 2]
```diff
@@ -29,16 +32,29 @@ export const Cluster = ({ cluster, match }) => {
...
+              icon={InfoCircleIcon}
+              iconStyle={{ color: globalInfoColor100.value }}
+              variant="large"
+              link="https://docs.openshift.com/container-platform/latest/support/getting-support.html"
+              linktext=" OpenShift documentation"
+            />
+          </Main>
+        ) : (
+          <Main>
+            <MessageState
```
Comment (gkarat) [inline src/Components/Cluster/Cluster.js:50]: Could, for the "404 case", `MessageState` can be utilized too? Did you try passing `text` param like:
```
...
text={<React.Fragment>{intl.formatMessage(messages.noRecsErrorDec}<a href="https://docs.openshift.com/container-platform/latest/support/getting-support.html">OpenShift documentation</a></React.Fragment>}
...

```

File: src/Components/MessageState/NoRecsState.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    {icon !== 'none' && (
+      <EmptyStateIcon className={iconClass} style={iconStyle} icon={icon} />
+    )}
+    <Title headingLevel="h5" size="lg">
+      {title}
+    </Title>
+    <EmptyStateBody style={{ marginBottom: '16px' }}>
+      {text}
+      <a href={link}>{linktext}</a>
```
Comment (gkarat) [inline src/Components/MessageState/NoRecsState.js:34]: Hmm.. WDYT about https://formatjs.io/docs/react-intl/components/#defaultrichtextelements?

File: src/Components/Cluster/Cluster.js (hunk 2 above)
Comment (Fewwy) [reply src/Components/Cluster/Cluster.js:50]: It worked! Haha! I will remove unnecessary component I created before

**Review by Fewwy — COMMENTED** [review]
Finished refactoring after discussion with @gkarat

File: src/Components/Cluster/Cluster.js [hunk 4]
```diff
@@ -29,16 +31,35 @@ export const Cluster = ({ cluster, match }) => {
...
-          <MessageState
-            title={intl.formatMessage(messages.noRecsError)}
-            text={intl.formatMessage(messages.noRecsErrorDesc)}
-            icon={ExclamationCircleIcon}
-            iconStyle={{ color: globalDangerColor100.value }}
-          />
-        </Main>
-      )}
+      {isError &&
+        (cluster.error?.status === 404 ? (
```
Comment (gkarat) [inline src/Components/Cluster/Cluster.js:35]: Can you access the `error` key from within https://github.com/RedHatInsights/ocp-advisor-frontend/pull/42/files#diff-ec15cd3d92c0a4ec195abe2378100993e0afe8634a744bcd259b507437d81488R24?

File: compiled-lang/en.json [hunk 5]
```diff
@@ -42,6 +42,8 @@
   "noRecommendationsDesc": "No known recommendations affect this cluster.",
   "noRecsError": "No recommendations available.",
```
Comment (gkarat) [inline compiled-lang/en.json:43]: Please, replace it to
```suggestion
  "noRecsError": "No recommendations available",
```

File: src/Components/Cluster/Cluster.js [hunk 6]
```diff
@@ -29,16 +31,35 @@ export const Cluster = ({ cluster, match }) => {
...
+            <MessageState
+              title={intl.formatMessage(messages.noRecsFoundError)}
+              text={
+                <React.Fragment>
+                  {intl.formatMessage(messages.noRecsFoundErrorDesc)}
+                  <a href="https://docs.openshift.com/container-platform/latest/support/getting-support.html">
+                    {' '}
+                    OpenShift documentation
+                  </a>
+                </React.Fragment>
```
Comment (gkarat) [inline src/Components/Cluster/Cluster.js:46]: Please, finish the body text with a dot

**Review by gkarat — CHANGES_REQUESTED** [review]

File: src/Components/Cluster/Cluster.js (hunk 6 above)
Comment (Fewwy) [reply src/Components/Cluster/Cluster.js:46]: Resolved

File: src/Components/Cluster/Cluster.js (hunk 4 above)
Comment (gkarat) [reply src/Components/Cluster/Cluster.js:35]: https://github.com/Fewwy/ocp-advisor-frontend/blob/ccxdev-5812-changing-error-status-responses/src/Components/Cluster/Cluster.js#L24

Comment (Fewwy) [reply src/Components/Cluster/Cluster.js]: Resolved

**Review by gkarat — APPROVED** [review]

**Pull request merged**